### PR TITLE
SCRUM-4137 include SGD background strain into unique ID generation

### PR DESCRIPTION
**Pull request by cmpich**

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/services/helpers/annotations/AnnotationUniqueIdHelper.java [hunk 1]
```diff
@@ -129,6 +125,11 @@ public static String getDiseaseAnnotationUniqueId(DiseaseAnnotation annotation)
 		if (annotation.getDiseaseGeneticModifierRelation() != null) {
 			uniqueId.add(annotation.getDiseaseGeneticModifierRelation().getName());
 		}
+		if(annotation instanceof GeneDiseaseAnnotation gda){
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/helpers/annotations/AnnotationUniqueIdHelper.java]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/helpers/annotations/AnnotationUniqueIdHelper.java [hunk 2]
```diff
@@ -129,6 +125,11 @@
...
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/helpers/annotations/AnnotationUniqueIdHelper.java]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'{' is not preceded with whitespace.

File: src/main/java/org/alliancegenome/curation_api/services/helpers/annotations/AnnotationUniqueIdHelper.java [hunk 3]
```diff
@@ -129,6 +125,11 @@
...
+			if(gda.getSgdStrainBackground() != null){
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/helpers/annotations/AnnotationUniqueIdHelper.java]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.

Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/services/helpers/annotations/AnnotationUniqueIdHelper.java]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'{' is not preceded with whitespace.

**Review by oblodgett — APPROVED** [review]

**Pull request merged**